bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -1255,6 +1259,30 @@ void ReplicaImp::onMessage<CheckpointMsg>(CheckpointMsg *msg) {
     onReportAboutLateReplica(msgSenderId, msgSeqNum);
   }
 }
+/**
+ * Is sent from a read-only replica
+ */
+template <>
+void ReplicaImp::onMessage<AskForCheckpointMsg>(AskForCheckpointMsg *msg) {
+  // metric_received_checkpoints_.Get().Inc(); // TODO [TK]
+
+  LOG_INFO(GL, "Node " << config_.replicaId << " received AskForCheckpoint message from node " << msg->senderId());
+
+  const CheckpointInfo &checkpointInfo = checkpointsLog->get(lastStableSeqNum);
+  CheckpointMsg *checkpointMsg = checkpointInfo.selfCheckpointMsg();
+
+  if (checkpointMsg == nullptr) {
```

Comment:
Does nothing?

Reply:
I'm not sure yet if it's right to uncomment, I'll consult Guy and decide.